hashintel/hash
# Standardize extraction of account id and page id from route

## 🌟 What is the purpose of this PR?

This PR provides an alternative solution to https://github.com/hashintel/hash/pull/456. It prevents account id from being undefined during SSR by not rendering app pages on the server. Frontend page files are converted to `kebab-case` given the opportunity.




## 🔍 What does this change?

- New hooks: `useRouteAccountInfo` and `useRoutePageInfo` (inspired by `CurrentWorkspaceContext`)

- Replacement of all `router.query["pageId|accountId"]` with new hooks. This improves type-safety and provides future-proofing. Using slugs instead of ids should be easier now.

- Page files were renamed `[account-slug]` and `[page-slug]` instead of `[accountId]` and `[pageId]`

- `_app.page.ts` is updated: server-side rendering is skipped because page and account ids are undefined and this may crash some components

     App UI often depends on `[account-slug]` and other query params. However, `router.query` is empty during server-side rendering for pages that don’t use `getServerSideProps`. By showing app skeleton on the server, we avoid UI mismatches during rehydration


- `kebab-case` is enforced for `"[frontend]/src/pages/**/*"` to speed-up future reviews. Related to https://github.com/hashintel/hash/pull/453




## ⚠️ Known issues

- Current ‘app skeleton’ is `null`. We should replace it with Hash logo or some loader. Alternatively, we can consider making all `/[account-slug]/*` pages server-rendered by adding `getServerSideProps`. However, this will significantly increase server load and create some additional edge cases for UI layouts.

## 🐾 Next steps

- Continue working on UI components for the entity editor

- Consider parsing `[account-slug]` and converting it to `accountId` inside `RouteAccountInfoProvider`




## 🔗 Related links





- https://github.com/hashintel/hash/pull/456
- https://github.com/hashintel/hash/pull/409#discussion_r836502710
- [Asana task](https://app.asana.com/0/1201959586244678/1202046726105145) _(internal)_

## 🛡 What tests cover this?



- Playwright, unit
- Manual

## ❓ How to test this?



1.  Checkout the branch / view the deployment
1.  Open account home page
1.  Navigate between a few pages
1.  Navigate to a page for a type and observe that the list of pages is still there.
